### DISCUSSION TITLE
How to properly catch error when endpoint fails

### DISCUSSION THREAD

**Discussion by venuziano** [sindresorhus/got#2245] [Q&A] [answered]
I'm using got to perform some endpoint calls between my APIs and would like to know how can I handle when an endpoint fails to fetch.

Currently, if I simulate a call with an invalid endpoint, **got** crashes my application:
![image](https://user-images.githubusercontent.com/15386984/229214928-0e28a23e-78c6-45f0-9b81-15c5d528134e.png)

I already tried to use some try-catchs, but them aren't triggered. Any idea what I'm missing here?

**Comment by sindresorhus** [comment, marked as answer]
Try-catch blocks should work fine with Got. Make sure you're using `await` and wrapping it properly:

```javascript
import got from 'got';

try {
	const response = await got('https://example.com/invalid-endpoint');
	console.log(response.body);
} catch (error) {
	console.error('Request failed:', error.message);
	console.error('Status code:', error.response?.statusCode);
}
```

If you're using promises without async/await, use `.catch()`:

```javascript
got('https://example.com/invalid-endpoint')
	.then(response => {
		console.log(response.body);
	})
	.catch(error => {
		console.error('Request failed:', error.message);
	});
```

If your app is still crashing, it might be because:
1. You're not awaiting the promise
2. You have an unhandled promise rejection elsewhere
3. You're using Got in a callback without proper error handeling
